os/fs/driver/mtd/m25px/m25px.c

```diff
@@ -413,10 +433,26 @@ static inline int m25p_readid(struct m25p_dev_s *priv)
 		} else if (capacity == M25P_M25P128_CAPACITY) {
 			/* Save the FLASH geometry */
 
-			priv->sectorshift = M25P_M25P128_SECTOR_SHIFT;
-			priv->nsectors = M25P_M25P128_NSECTORS;
-			priv->pageshift = M25P_M25P128_PAGE_SHIFT;
-			priv->npages = M25P_M25P128_NPAGES;
+			if (manufacturer == WINBOND_SFLASH_ID &&
```

Comment:
Line 377,
```
	if (manufacturer == M25P_MANUFACTURER && memory == M25P_MEMORY_TYPE) {
```
This covers here, I think this never calls. Could you check it?

Reply:
As you know, it is to support winbond flash inside M25px driver.
M25P_MANUFACTURER  & MEMORY_TYPE are configurable. 
Lets say it is set to M25P128 chip, manufacture = C2, memory = 20, then for capacity 128 we need to M25P128 configurations. 
For W25Q128 case, we explicitly check for winbond id & mem type and use W25Q128 configurations.

Reply:
I don't understand what you mean. Anyway, Line 377 and line 436 use same variables, `manufacturer` and `memory`. How do they have different value? That is not a bit comparision.

Reply:
Oh ok. I understood. Line 377 is a comparision between read values from SPI and configured values but line 436 is a comparision between read values and winbond values.
It looks little strange on readability. Because of definitions of M25P_MANUFACTURER and WINBOND_SFLASH_ID, I thought they are definitely different values.

If we support w25 in m25 feature, do we need to keep w25.c?

Reply:
Yes. This supports only a single winbond flash in m25px driver whereas w25.c is for all winbond family flash's.
As you know, this requirement came from one of our issue in our private repo where we have used m25px driver for winbond flash(for some reason).  For code sync, I had to push this patch here.